ablySpec/RealtimeClientChannel.swift

```diff
@@ -13,6 +13,32 @@ class RealtimeClientChannel: QuickSpec {
     override func spec() {
         describe("Channel") {
 
+            // RTL4
+            describe("attach") {
+
+                // RTL4a
+                it("if already ATTACHED or ATTACHING nothing is done") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    var errorInfo: ARTErrorInfo?
+                    let channel = client.channel("test")
+
+                    errorInfo = channel.attach()
```

Comment:
@ricardopereira There's no need to open issues for API inconsistencies, as I'm reviewing everything. Do it if you feel like it, but I'm on it anyway!